src/openrct2-ui/windows/LandRights.cpp

```diff
@@ -295,9 +295,9 @@ static void window_land_rights_tool_update_land_rights(ScreenCoordsXY screenCoor
     map_invalidate_selection_rect();
     gMapSelectFlags &= ~MAP_SELECT_FLAG_ENABLE;
 
-    CoordsXY mapTile = screen_get_map_xy(screenCoords, nullptr);
+    auto mapTile = screen_get_map_xy(screenCoords, nullptr);
 
-    if (mapTile.x == LOCATION_NULL)
+    if (!mapTile)
```

Comment:
We always compare with `nullptr`.

Reply:
This isn't a nullptr it's a optional. This is the way we have done this elsewhere

Reply:
I wonder if we should. Before optionals came along, we used explicit comparisons for anything that wasn't a boolean. I think explicit comparisons are clearer.

Reply:
A `std::optional` has well-defined behaviour through the type system, though. I think this does add semantic value over a simple nullptr.

Reply:
Strictly speaking it is a boolean operation, see https://en.cppreference.com/w/cpp/utility/optional/operator_bool

Reply:
Since it was marked as resolved, I'm assuming nothing needs to be done.